src/test/coins_tests.cpp

```diff
@@ -559,21 +561,58 @@ BOOST_AUTO_TEST_CASE(ccoins_serialization)
 }
 
 const static COutPoint OUTPOINT;
-const static CAmount SPENT = -1;
-const static CAmount ABSENT = -2;
-const static CAmount FAIL = -3;
-const static CAmount VALUE1 = 100;
-const static CAmount VALUE2 = 200;
-const static CAmount VALUE3 = 300;
-const static char DIRTY = CCoinsCacheEntry::DIRTY;
-const static char FRESH = CCoinsCacheEntry::FRESH;
-const static char NO_ENTRY = -1;
-
-const static auto FLAGS = {char(0), FRESH, DIRTY, char(DIRTY | FRESH)};
-const static auto CLEAN_FLAGS = {char(0), FRESH};
-const static auto ABSENT_FLAGS = {NO_ENTRY};
-
-static void SetCoinsValue(CAmount value, Coin& coin)
+constexpr CAmount SPENT {-1};
+constexpr CAmount ABSENT{-2};
+constexpr CAmount VALUE1{100};
+constexpr CAmount VALUE2{200};
+constexpr CAmount VALUE3{300};
+
+struct CoinEntry {
+    enum class State { CLEAN, DIRTY, FRESH, DIRTY_FRESH };
+
+    const CAmount value;
+    const State state;
+
+    constexpr CoinEntry(const CAmount v, const State s) : value{v}, state{s} {}
```

Comment:
Using `const` for value type parameters (`CAmount is a typedef of `int64_t`), when a function doesn't contain any logic is noisy to me and not used in the codebase at large.
```suggestion
    constexpr CoinEntry(CAmount v, State s) : value{v}, state{s} {}
```

It is useful in cases like this where the function is actually doing something:
https://github.com/bitcoin/bitcoin/blob/ebe4cac38bf6c510b00b8e080acab079c54016d6/src/uint256.cpp#L20-L44

Same goes for `ToState`, `SetCoinsValue`, `SingleEntryCacheTest`, `CheckAccessCoin` etc.

Otherwise you might as well add `const` to `AddFlags(uint8_t flags, ...` etc?

Reply:
I prefer `const` here, my IDE was also asking for them, I don't mind adding them here.
Being `noisy` when `a function doesn't contain any logic` doesn't sound like a blocker to me :)

Reply:
This issue is a non-nit for me as a general pattern as it will affect diffs & reviews going forward.

We've got a ratio of 5/212 or roughly 1:40 for this codebase. I'm surprised that the IDE is encouraging that.

What is you rationale for not applying it to `AddFlags`? Did the specific version of your IDE not tell you to do so? Does your IDE have a documented rationale?

If there is a clear rationale, ideally with a statistical sample of % bugs discovered from changing to `const` value type parameters, I'm open to that. But the 1:40 ratio weighs heavily.

Reply:
We can unify these in a separate PR, I don't mind, I added these since the linters were complaining - and I was fine with either option since it doesn't affect the body of the function signal-to-noise-ratio-wise.

Reply:
9edbe5a48fe9a2c0d364882fcb7d44ae6aac6d24 before the latest push didn't have these `const` params. I think it was a mistake to add them in the latest push and would prefer them rolled back.

Another alternative is to discuss my samples of ratios of `const` value type params or why you didn't apply it to `AddFlags`.

Sorry to be annoying about this but for some reason it really annoys me. :)
Will take a timeout and see if I can let go of this, if you still insist.